### ISSUE TITLE
Multiple assignment in conditionals triggers SyntaxError

### ISSUE THREAD

**Issue by deepj** [oracle/truffleruby#1513]
Ruby 2.4 changed a behavior and in case of multiple assignment in conditionals no SyntaxtError is thrown

To reproduce:

```ruby
if (x, y = [1,2]) then 'passed' else 'failed' end
```

Error:
```
SyntaxError: (irb):23: multiple assignment in conditional
	from ~/.rubies/truffleruby-1.0.0-rc10/bin/irb:29:in `<main>'
```

Expected result:

```
(irb):1: warning: found = in conditional, should be ==
=> "passed"
```

**Comment by eregon** [comment]
The warning is only because the RHS is a simple literal like `[1,2]`.
This doesn't warn on MRI:
```
$ ruby -e 'if (a,b=[1,2.even?]); p a; end'
1

$ truffleruby -e 'if (a,b=[1,2.even?]); p a; end'
truffleruby: -e:1: multiple assignment in conditional (SyntaxError)
```
(from #1912)

**Comment by norswap** [comment]
Fixed by 48f936ea539e340d5742abb607aff8733c7f19ea
